unified-installer/astra-unified-installer.sh

```diff
@@ -1117,12 +1179,177 @@ is_astra_registry() {
+        cat <<EOF > "$kustomization_file"
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+patches:
+transformers:
+generatorOptions:
+  disableNameSuffixHash: true
+secretGenerator:
+EOF
+        logdebug "$kustomization_file: OK"
+    fi
+
+    if [ "$add_as_resource_to_top_kustomization_file" == "true" ]; then
+        local -r top_kustomization_file="$__GENERATED_KUSTOMIZATION_FILE"
+        local -r base_dir="$(basename "$kustomization_dir")"
+        insert_into_file_after_pattern "$top_kustomization_file" "resources:" "- ./$base_dir"
+        logdebug "$top_kustomization_file: added resource entry for resource at path '$base_dir'"
+    fi
+}
+
+add_astra_registry_secret_to_kustomization() {
+    local -r kustomization_file="${1}"
+    local -r namespace="${2}"
+    local -r registry="${3}"
+    [ -z "$kustomization_file" ] && fatal "no kustomization file given"
+    [ ! -f "$kustomization_file" ] && fatal "kustomization file '$kustomization_file' does not exist"
+    [ -z "$namespace" ] && fatal "no namespace given"
+    [ -z "$registry" ] && fatal "no registry given"
+
+    local -r account_id="$ASTRA_ACCOUNT_ID"
+    local -r api_token="$ASTRA_API_TOKEN"
+    local -r pull_secret="$__ASTRA_REGISTRY_IMAGE_PULL_SECRET"
+    [ -z "$account_id" ] && fatal "no account_id found"
+    [ -z "$api_token" ] && fatal "no api_token found"
+    [ -z "$pull_secret" ] && fatal "no pull_secret found"
+
+    local -r encoded_creds=$(echo -n "$account_id:$api_token" | base64)
+
+    # Add secret generator
+    insert_into_file_after_pattern "$kustomization_file" "secretGenerator:" \
+'- name: "'"$pull_secret"'"
+  namespace: "'"$namespace"'"
+  type: kubernetes.io/dockerconfigjson
+  literals:
+  - |
+    .dockerconfigjson={
+      "auths": {
+        "'"$registry"'": {
+          "username": "'"$account_id"'",
+          "password": "'"$api_token"'",
+          "auth": "'"$encoded_creds"'"
+        }
+      }
+    }'
+    logdebug "$kustomization_file: added secret '$pull_secret' to namespace '$namespace'"
+}
+
+add_api_token_secret_to_kustomization() {
+    local -r kustomization_file="${1}"
+    local -r namespace="${2}"
+    [ -z "$kustomization_file" ] && fatal "no kustomization file given"
+    [ ! -f "$kustomization_file" ] && fatal "kustomization file '$kustomization_file' does not exist"
+    [ -z "$namespace" ] && fatal "no namespace given"
+
+    local -r api_token="$ASTRA_API_TOKEN"
+    local -r pull_secret="$__ASTRA_API_TOKEN_PULL_SECRET"
+    [ -z "$api_token" ] && fatal "no api_token found"
+    [ -z "$pull_secret" ] && fatal "no pull_secret found"
+
+    # Add API token secret
+    insert_into_file_after_pattern "$kustomization_file" "secretGenerator:" \
+"- name: \"${pull_secret}\"
+  namespace: \"${namespace}\"
+  literals:
+  - apiToken=\"${api_token}\""
+    logdebug "$kustomization_file: added secret '$pull_secret' to namespace '$namespace'"
+}
+
+add_imagepullsecret_patch_to_kustomization() {
```

Comment:
This is the 3rd part of `step_kustomize_global_pull_secret_if_needed`. This piece is used for both the connector and trident kustomization files, and is for setting the **imagePullSecrets** in the operator deployments.